packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -181,12 +187,17 @@ export class QueuedRequestController extends BaseController<
       (_, patch) => {
         patch.forEach(({ op, path }) => {
           if (
-            (op === 'replace' || op === 'add') &&
             path.length === 2 &&
             path[0] === 'domains' &&
             typeof path[1] === 'string'
           ) {
-            this.#flushQueueForOrigin(path[1]);
+            const origin = path[1];
+            this.#flushQueueForOrigin(origin);
+            // When a domain is removed from SelectedNetworkController, its because of revoke permissions.
+            // Rather than subscribe to the permissions controller event in addition to the selectedNetworkController ones, we simplify it and just handle remove on this event alone.
+            if (op === 'remove' && origin === this.#originOfCurrentBatch) {
+              this.#clearPendingConfirmations();
```

Comment:
So `clearPendingConfirmations` will only clear confirmations in the current 'unqueued' batch?